### PR TITLE
Remove adjustment callbacks

### PR DESCRIPTION
**Pull request by mbj**
Extraction from #398

### CODE REVIEW

File: core/spec/models/spree/order/checkout_spec.rb [hunk 1]
```diff
@@ -76,7 +76,7 @@
     end
 
     context "from address" do
-      let!(:line_item) { create(:line_item, price: 10, adjustment_total: -10) }
+      let!(:line_item) { create(:line_item, price: 10) }
```
Comment (mbj) [inline core/spec/models/spree/order/checkout_spec.rb]: Just setting an arbitrary `adjustment_total` without creating any adjustment is invalid state. The code before silently cleaned up that invalid state, but I'd rather not specify such a thing.

**Comment by mbj** [comment]
@dkubb This is ready for review.

File: core/spec/models/spree/order/checkout_spec.rb (hunk 1 above)
Comment (dkubb) [inline core/spec/models/spree/order/checkout_spec.rb]: Would preventing `adjustment_total` from being mass assignable eliminate this as a possibility?

Comment (mbj) [inline core/spec/models/spree/order/checkout_spec.rb]: It needs to be mass assigned from the `ItemAdjustments`, but in the long run its likely a good idea. Nothing I'd do in this PR.

Comment (dkubb) [inline core/spec/models/spree/order/checkout_spec.rb]: Sure, although I wonder if we should audit all the models and remove mass assignability from all internally computed attributes.

Comment (mbj) [inline core/spec/models/spree/order/checkout_spec.rb]: Definitively, but not this PR.

**Pull request merged**